File: app/Models/Hazard.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Hazard extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'user_name',
        'hazard_category',
        'hazard_description',
        'latitude',
        'longitude',
        'location_name',
        'device_info',
        'reported_at',
    ];

    protected $casts = [
        'reported_at' => 'datetime',
    ];
}

File: android/app/src/main/java/com/example/guglemap/models/Hazard.kt
package com.example.guglemap.models

import com.google.gson.annotations.SerializedName

data class Hazard(
    @SerializedName("id")
    val id: Int? = null,
    
    @SerializedName("user_name")
    val userName: String,
    
    @SerializedName("hazard_category")
    val hazardCategory: String,
    
    @SerializedName("hazard_description")
    val hazardDescription: String,
    
    @SerializedName("latitude")
    val latitude: Double,
    
    @SerializedName("longitude")
    val longitude: Double,
    
    @SerializedName("location_name")
    val locationName: String? = null,
    
    @SerializedName("device_info")
    val deviceInfo: String? = null,
    
    @SerializedName("reported_at")
    val reportedAt: String? = null
)
